src/ERC20DelegatesUpgradeable.sol

```diff
@@ -116,6 +105,27 @@ abstract contract ERC20DelegatesUpgradeable is
         _moveDelegateVotes(delegates(from), delegates(to), amount);
     }
 
+    /// @dev Returns the current total supply of votes.
+    function _getTotalSupply() internal view returns (uint256) {
+        ERC20DelegatesStorage storage $ = _getERC20DelegatesStorage();
+        return $._totalVotingPower;
+    }
+
+    /// @dev Must return the voting units held by an account.
+    function _getVotingUnits(address account) internal view returns (uint256) {
+        return balanceOf(account);
+    }
+
+    /// @dev Moves voting power when tokens are transferred.
+    /// @dev Emits a {IDelegates-DelegateVotesChanged} event.
+    function _update(address from, address to, uint256 value) internal virtual override {
+        super._update(from, to, value);
+        // No check of supply cap here like in OZ implementation as MORPHO has a 1B total supply cap.
+        _transferVotingUnits(from, to, value);
+    }
+
+    /* PRIVATE */
```

Comment:
I wonder whether we should put private to internal

Reply:
No strong opinion on this